sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2205,6 +2205,12 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_USE_HASH_ON_MAPTYPE = buildConf("spark.sql.legacy.useHashOnMapType")
```

Comment:
Maybe, `allowHashOnMapType`?

Reply:
ah "allow" is more precise. @iRakson feel free to send a followup to address all the comments.